cmd/collector/collector.go

```diff
@@ -60,6 +60,19 @@ func initLoggingToFile(fs *pflag.FlagSet) {
 	}
 }
 
+func signalHandler(stopCh chan struct{}) {
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+
+	for {
+		select {
+		case <-signalCh:
+			close(stopCh)
+			return
+		}
+	}
+}
```

Comment:
Just curious, if we have only one case, do we still need to use select?
For example, does the following suit our need?

```
func signalHandler(stopCh chan struct{}) {
	signalCh := make(chan os.Signal, 1)
	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
	<-signalCh
	close(stopCh)
}
```
Thanks.

Reply:
Yes, you are right. We can simplify the code.
Taking inspiration from Antrea applications, I defined signalCh with capacity 2 to handle the case if a second interrupt/termination comes through. In that case, program exits forcefully.

Reply:
Yes, I meant the go routine. Thanks for the explanation.